pkg/kubelet/cm/event_dispatcher.go

```diff
@@ -41,6 +41,14 @@ type EventDispatcher interface {
 	// PostStopPod is invoked after all of a pod's containers have permanently
 	// stopped running, but before the pod sandbox is destroyed.
 	PostStopPod(cgroupPath string) error
+
+	// PreStartContainer is invoked before the container is created.
+	PreStartContainer(podName, containerName string) (*lifecycle.EventReply, error)
```

Comment:
@ConnorDoyle Is podName and container name enough ?  i.e we have 2 containers in pod and we want to have 2 cpus for container "busy1" and 3 cpus for container "busy2" (separate OIR's entries in podSpec for each container). Then with `PreStartPod` call we setup limit in cpuset for podSandbox group assigning in total 5 cpus.

Then when we call `PreStartContainer` (i.e for busy1 container) we want to chose 2 cpus (from the 5 already assigned).
How will isolator know the 2 cpus should be chosen and not 3 or 4 ?

One option would be for the isolator to either save podSpec from `PreStartPod` Event, or request podSpec directly from k8s API but those approaches put more complexity on isolator.

IMO If we change `PreStartContainer` to sent container name and podSpec or containerConfig the logic in isolator will be simpler. What do you think ?